### PR TITLE
[azure-devops] 1st commit

### PR DESCRIPTION
**Pull request by joalmeid**
1st PR into the azure-devops branch! Requesting some analysis, before continuing.

### CODE REVIEW

**Comment by jordanh** [comment]
Well _this_ is exciting @joalmeid! @mattkrick and I meet tomorrow at 5p UTC+1 and we'll set aside some time to review this and get some face to face time scheduled with you. We're excited to bring this over the finish line with you

File: packages/server/graphql/queries/helpers/suggestedIntegrationHelpers.ts [hunk 1]
```diff
@@ -114,8 +119,24 @@ export const getPermsByTaskService = async (
       .default(null)
   ])
 
+  //TODO: not ideal. Need to understand rethink schema.
+  const [azureDevopsAuths] = await Promise.all([
+    dataLoader.get('azureDevopsAuthByUserId').load(userId),
+    r
```
Comment (mattkrick) [inline packages/server/graphql/queries/helpers/suggestedIntegrationHelpers.ts]: can remove this line. way back we tried keeping all integrations in a single `Provider` table, but as it turns out each provider is so different that we broke them out into their own tables.  Still need to rename `Provider` to `GitHub` or similar.

File: packages/server/graphql/types/AzureDevopsAuth.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      type: new GraphQLNonNull(GraphQLBoolean)
+    },
+    accessToken: {
+      description:
+        'The access token to Azure Devops, useful for 1 hour. null if no access token available',
+      type: GraphQLID,
+      resolve: async ({teamId, userId}, _args, {authToken, dataLoader}) => {
+        const viewerId = getUserId(authToken)
+        if (viewerId !== userId) return null
+        return dataLoader.get('freshAzureDevopsAccessToken').load({teamId, userId: viewerId})
```
Comment (mattkrick) [inline packages/server/graphql/types/AzureDevopsAuth.ts]: there will likely be some nuance here. Jira provides a JWT with a 30 minute expiration. If devops gives a longer timewindow we could probably ignore the refresh entirely

**Comment by joalmeid** [comment]
Hey @mattkrick !
Thanks for the review. I already have a auth implementation. I'll add code changes for your comments. 
I'll then send updates through here/email.

**Comment by joalmeid** [comment]
Closed this PR to continue on #3464 -

**Pull request closed without merging**